test/mocks/xpcom.js

```diff
@@ -0,0 +1,16 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+window.Components = {
+  utils: {
+    "import": function() {},
+  },
+};
+
+window.Services = {
+  telemetry: {
+    registerEvents() {},
+    recordEvent() {},
```

Comment:
(for the future, since this kind of comment should also have been made long ago elsewhere)

this can be a good place to keep a recording log, and  verify the calls that are expected match what was captured.

Reply:
I don't think I'm doing it right now, but generally I just leave the functions in our mocks empty (unless we need some functionality for a test), and then use Sinon to watch for when they're called in the relevant test so I can make sure they were called with the right args.